site/en/docs/extensions/reference/offscreen/index.md

```diff
@@ -2,4 +2,37 @@
 api: offscreen
 ---
 
-<!-- Intentionally blank -->
+## Manifest
+You must declare the "offscreen" permission in the [extension manifest][1] to use the Offscreen API. For example:
+
+```json
+{
+  "name": "My extension",
+  ...
+  "permissions": [
+    "offscreen"
+  ],
+  ...
+}
+```
+Pages loaded as offscreen documents are handled differently from other types of extensions pages. The extension's permissions carry over to offscreen documents, but extension API access is heavily limited. Currently, an offscreen document can only use the `chrome.runtime` APIs to send and receive messages; all other extension APIs are not exposed. Other notable differences between offscreen documents and normal pages are as follows:
+
+* Offscreen document's URL must be a static HTML file bundled with the extension.
+* Offscreen documents cannot be focused.
+* Offscreen documents cannot have their `opener` property set using the [`chrome.windows` API](docs/extensions/reference/windows/) method `windows.setSelfAsOpener`.
+
+## Reasons
+Reasons, listed [below](/docs/extensions/reference/offscreen/#type-Reason), are set upon  document creation to determine the document's lifespan. Currently, the AUDIO_PLAYBACK reason has specialized lifetime enforcement (CreateAudioLifetimeEnforcer). All others use generic idle detection   (CreateEmptyEnforcer).
```

Comment:
Nit: Thanks for adding this, but can we link to https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/api/offscreen/audio_lifetime_enforcer.h and https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/api/offscreen/lifetime_enforcer_factories.cc. That at least gives an indication of the code we're referencing and a place to read more :)

Reply:
@jpmedley I haven't seen direct links to chromium source yet in other docs, let me know if this is within good practice.

Reply:
Linking to source code is an anti-pattern on any platform. 
* It avoids using it as a crutch for things that aren't documented, but should be. 
* Avoids providing information in a programming language that the audience may not be familiar with. 

Please do a follow-up PR to add these to our documents. As we progress over the next year, we should have less and less reason to do patches like this.

Reply:
The thing that worried me was that mentioning `CreateEmptyEnforcer` is completely useless to developers without telling them what it refers to. But happy for this to be a follow-up 👍

Reply:
It should be a follow up.